inst/pages/exercises.qmd

```diff
@@ -1805,6 +1805,227 @@ mat <- assay(tse, "standardize")
+```
+
+2. Calculate Alpha Diversity (Shannon Index), i.e. the dependent variable.
+
+```{r}
+#| label: extra-regressionchart-step2
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: true
+
+tse <- addAlpha(tse)
+```
+
+Now let's define the covariates i.e. the independent variables.
+
+3. First, fit linear model using a discrete variable, say "SampleType", where 
+Shannon diversity is the response variable, and SampleType is the predictor. 
+
+"SampleType" represents the type or source of the microbiome sample. It 
+categorizes each sample based on the environment or biological material from
+which it was collected.
+
+```{r}
+#| label: extra-regressionchart-step3
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: true
+
+model <- lm(shannon ~ SampleType, data = colData(tse))
+```
+
+4. Visualize the results using a boxplot. This plot helps in understanding the
+distribution of diversity across discrete sample types.
+
+```{r}
+#| label: extra-regressionchart-step4
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: true
+
+plotColData(tse, x = "SampleType", y = "shannon", color_by = "SampleType") +
+  geom_boxplot() +
+  theme_minimal()
+```
+
+What do we interpret from the above regression analysis?
+
+i. The boxplot generated shows the distribution of Shannon diversity across
+different sample types. If the boxes are well-separated and there are 
+significant differences in medians, it indicates that SampleType has a strong
+effect on diversity. This would be supported by significant p-values in the
+regression summary.
+ii. The p-value associated with each coefficient tells us whether the effect
+of that sample type on the Shannon diversity index is statistically
+significant. A common threshold for significance is 0.05.
+iii. R-squared: This value indicates the proportion of the variance in the
+Shannon diversity index that is explained by the sample type. A higher R-squared
+value indicates a better fit of the model.
+
+Now, let's check some linear modeling assumptions :
+  
+```{r}
+#| label: sub-checking-modeling-assumptions
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: true
+
+# - Residual diagnostics to visually check for normality and homoscedasticity.
+# - Durbin-Watson Test for independence of residuals
+
+par(mfrow = c(2, 2))
+plot(model)
+dw_result <- dwtest(model)
+print(dw_result)
+```
+
+Here, 
+
+i. Residual diagnostics (normality and homoscedasticity):
+
+a. The Residuals vs Fitted plot provides insight into the relationship 
+between residuals and fitted values. Ideally, the residuals should be randomly
+scattered around the horizontal line (residuals = 0) without any discernible 
+pattern. This indicates that the model’s assumption of linearity and homoscedasticity
+(constant variance of residuals) is likely met. If you observe any systematic patterns
+(such as a funnel shape), it may suggest heteroscedasticity, meaning the variance of
+residuals is not constant.
```

Comment:
You could add question: "Do the assumptions hold?". And then add answer with folded text: https://quarto.org/docs/authoring/callouts.html